providers/implementations/rands/drbg.c

```diff
@@ -311,75 +294,44 @@ static const OSSL_LIB_CTX_METHOD drbg_nonce_ossl_ctx_method = {
 };
 
 /* Get a nonce from the operating system */
-static size_t prov_drbg_get_nonce(PROV_DRBG *drbg,
-                                  unsigned char **pout,
-                                  int entropy, size_t min_len, size_t max_len)
+static size_t prov_drbg_get_nonce(PROV_DRBG *drbg, unsigned char **pout,
+                                  size_t min_len, size_t max_len)
 {
     size_t ret = 0, n;
-    RAND_POOL *pool;
     unsigned char *buf = NULL;
     OSSL_LIB_CTX *libctx = PROV_LIBCTX_OF(drbg->provctx);
     PROV_DRBG_NONCE_GLOBAL *dngbl
         = ossl_lib_ctx_get_data(libctx, OSSL_LIB_CTX_DRBG_NONCE_INDEX,
                                 &drbg_nonce_ossl_ctx_method);
     struct {
-        void *instance;
+        void *drbg;
         int count;
     } data;
-    
+
     if (dngbl == NULL)
         return 0;
 
-    if (drbg->parent != NULL) {
-        if (drbg->parent_nonce != NULL) {
-            n = drbg->parent_nonce(drbg->parent, NULL, 0, drbg->min_noncelen,
-                                   drbg->max_noncelen);
-            if (n > 0 && (buf = OPENSSL_malloc(n)) != NULL) {
-                ret = drbg->parent_nonce(drbg->parent, buf, 0,
-                                         drbg->min_noncelen,
-                                         drbg->max_noncelen);
-                if (ret == n) {
-                    *pout = buf;
-                    return ret;
-                }
-                OPENSSL_free(buf);
+    if (drbg->parent != NULL && drbg->parent_nonce != NULL) {
+        n = drbg->parent_nonce(drbg->parent, NULL, 0, drbg->min_noncelen,
+                               drbg->max_noncelen);
+        if (n > 0 && (buf = OPENSSL_malloc(n)) != NULL) {
+            ret = drbg->parent_nonce(drbg->parent, buf, 0,
+                                     drbg->min_noncelen, drbg->max_noncelen);
+            if (ret == n) {
+                *pout = buf;
+                return ret;
             }
+            OPENSSL_free(buf);
```

Comment:
should this be a clear_free?

Reply:
The get failed, I don't see that it needs to be cleared in this case.